.gitignore

```diff
@@ -1,3 +1,29 @@
+# Ignore directories generated by Composer
+/drush/contrib/
+/vendor/
+/docroot/core/
+/docroot/modules/contrib/
+/docroot/themes/contrib/
+/docroot/profiles/contrib/
+/docroot/libraries/
+
+# Ignore sensitive information
+/docroot/sites/*/settings.php
```

Comment:
I think no need to add settings.php here as we usually update it according to the used hosting like Acquia.